rest-server/src/util/error.js

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+// documentation files (the 'Software'), to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+const httpErrors = require('http-errors');
+const statuses = require('statuses');
+
+const createError = exports = module.exports = (status, code, message) => {
```

Comment:
2 questions here:
- What does `createError = exports = module.exports` mean?
- What is the difference between `module.exports` and `exports`?

Reply:
Since this module will be used by other developers lots of times, I concerned giving more hints to editor to help developer using it.

The simple answer is:
- `module.exports = () => {}` is the general form of export a value as module itself, and can be imported by `const createError = require('error');`
- `exports.unknown = () => {}` is the general form of export a value as module's field, and can be imported by `const createUnknownError = require('error').unknown;`

These two general forms can be easily detected by editors and give helps to autocomplete/intellisense.

When the commonJS module is initialized, module loader will assign an empty object to both `exports` and `module.exports`, and use the value of `module.exports` as module exports.
In other words. `exports` is just an alias of `module.exports`. When `module.exports` is overwritten, exports still reference the original `module.exports` object assigned by module loader, which will not be exported. So simply assign value to `exports.unknown` will not make the value exported, though it can be detected by editors. After assign the newly exported `module.exports` value to `exports`, any field assigned to `exports` can be exported in normal.

However, since `d.ts` is written, these hints only helps editors without supports of TypeScript but able to detect exports of commonJS module. Such as legacy WebStorm. (Didn't use it for a while)